### PR TITLE
Add add_hotkey_event() to context.c

### PR DESCRIPTION
**Pull request by xVin**
targets #400

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Merge conflicts would occur. Please rebase.

**Comment by xVin** [comment]
Trying to... please see irc query.

**Comment by xVin** [comment]
Pls review

File: src/context.c [hunk 1]
```diff
@@ -176,3 +180,77 @@ func_exec(
...
+    struct ws_string* str;
+
+    stack += 2; //Ignoring object and command name
+
+    if (ws_value_get_type(&stack->value) != WS_VALUE_TYPE_STRING) {
+        return -EINVAL;
+    }
+
+    vstr = &stack->string;
+    str = ws_value_string_get(vstr);
```
Comment (matthiasbeyer) [inline src/context.c]: You can merge this and the line above.

File: src/context.c [hunk 2]
```diff
@@ -176,3 +180,77 @@ func_exec(
...
+    }
+
+    size_t i = 0;
+    uint16_t arr[len];
+    ITERATE_ARGS_TYPE(it, stack, v, int) {
+        arr[i] = (uint16_t) v;
+        ++i;
+    }
+
+    return ws_hotkey_add(str, arr, len);
```
Comment (matthiasbeyer) [inline src/context.c]: Return value should be on stack as well. (maybe as boolean type...)

File: src/context.c [hunk 3]
```diff
@@ -176,3 +180,77 @@ func_exec(
...
+    }
+
+    size_t i = 0;
+    uint16_t arr[len];
+    ITERATE_ARGS_TYPE(it, stack, v, int) {
+        arr[i] = (uint16_t) v;
+        ++i;
+    }
+
+    return ws_hotkey_remove(arr, len);
```
Comment (matthiasbeyer) [inline src/context.c]: Return value should be on stack, maybe as boolean.

**Comment by manuelmessner** [comment]
ACK

**Comment by neithernut** [comment]
Nice, but you might be better off pulling out the code where you assemble the array into a static helper function.

**Comment by xVin** [comment]
Hm, but that would be pulling out only 3 lines of code that are used in 2 methods, is that really worth it or am I overlooking sth?

**Comment by neithernut** [comment]
Mh.. I thought you can also pull out hte _counting_ of items into the same method, but I realized the advantages are nullified by the fact that you'd have to allocate the memory on the heap then...
So,... ACK.

**Pull request merged**